hoomd/md/IntegratorTwoStep.cc

```diff
@@ -112,6 +112,21 @@ void IntegratorTwoStep::update(uint64_t timestep)
         method->includeRATTLEForce(timestep + 1);
         }
 
+#ifdef ENABLE_MPI
+    if (m_sysdef->isDomainDecomposed())
+        {
+        // Force a 2nd ghost update of rigid body consituent particles to update
+        // constituent velocities
+        m_comm->beginUpdateGhosts(timestep + 1);
+        m_comm->finishUpdateGhosts(timestep + 1);
+        updateRigidBodies(timestep + 1);
+        }
+    else
+#endif
+        {
+        // Update rigid body constituent particles in serial simulations.
+        updateRigidBodies(timestep + 1);
+        }
```

Comment:
Since `updateRigidBodies` is called on both code paths, you can just use an `if`
```suggestion
    // update rigid body constituent particles at end of step
#ifdef ENABLE_MPI
    if (m_sysdef->isDomainDecomposed())
        {
        m_comm->beginUpdateGhosts(timestep + 1);
        m_comm->finishUpdateGhosts(timestep + 1);
        }
#endif
        updateRigidBodies(timestep + 1);
```

Please also remove the comment below. It is no longer needed because we are forcing this update.